pkg/kubelet/kubelet.go

```diff
@@ -2508,15 +2505,16 @@ func (kl *Kubelet) GetKubeletContainerLogs(podFullName, containerName string, lo
 		podStatus = pod.Status
 	}
 
-	if err := kl.validatePodPhase(&podStatus); err != nil {
-		// No log is available if pod is not in a "known" phase (e.g. Unknown).
-		return fmt.Errorf("Pod %q in namespace %q : %v", name, namespace, err)
+	switch podStatus.Phase {
+	case api.PodRunning, api.PodSucceeded, api.PodFailed:
+	default:
+		return fmt.Errorf("pod %q is in an unknown state and logs are not available", pod.Name)
```

Comment:
Same code as before, but agree the message should be changed.  I briefly debated having the kubelet hold the request until the pod became available (because in practice most clients want to do something like that) but it would require a lot of extra timeout logic that we can add in the future.